### PR TITLE
Feature/G030CxT

### PR DESCRIPTION
**Pull request by hitech95**
This PR fixes/implements the following **bugs/features**

* [x] G030CxT variants

The variant G030CxT was missing both default clock and the linker scripts.

**Validation**

* This has been tested with the validation example as stated in the wiki.
* Tested in Platform.io other than Arduino IDE
* A board with such MCU is running this code since yesterday (so far no locks):
```cpp
#include <Arduino.h>

#define LED PC7

void setup()
{
  // put your setup code here, to run once:
  // initialize digital pin LED_BUILTIN as an output.
  pinMode(LED, OUTPUT);
  Serial.begin(115200);
}

void loop()
{
  char hw[] = "Hello World!\0";
  size_t i;
  for (i = 0; i < strlen(hw); i++)
  {
    // put your main code here, to run repeatedly:
    digitalWrite(LED, HIGH); // turn the LED on (HIGH is the voltage level)
    delay(50);               // wait
    digitalWrite(LED, LOW);  // turn the LED off by making the voltage LOW
    delay(50);               // wait
    Serial.print(hw[i]);
  }
  delay(1000); // wait for a second
  Serial.println("...");
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -293,6 +293,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
+| :green_heart: | STM32G030C6<br>STM32G030C8 | Generic Board | *2.1.0* |
```
Comment (fpistm) [inline README.md:296]:
```suggestion
| :yellow_heart: | STM32G030C6<br>STM32G030C8 | Generic Board | **2.2.0** |
```

File: variants/STM32G0xx/G030C(6-8)T/generic_clock.c [hunk 2]
```diff
@@ -20,8 +20,41 @@
...
+  RCC_OscInitStruct.PLL.PLLN = 8;
+  RCC_OscInitStruct.PLL.PLLP = RCC_PLLP_DIV2;
+  RCC_OscInitStruct.PLL.PLLR = RCC_PLLR_DIV2;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK)
+  {
+    Error_Handler();
+  }
+  /** Initializes the CPU, AHB and APB buses clocks
+  */
+  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK|RCC_CLOCKTYPE_SYSCLK
```
Comment (fpistm) [inline variants/STM32G0xx/G030C(6-8)T/generic_clock.c:48]: Apply astyle on this file else CI will not be passed.
https://github.com/stm32duino/wiki/wiki/Astyle

**Review by fpistm — CHANGES_REQUESTED** [review]

File: variants/STM32G0xx/G030C(6-8)T/generic_clock.c (hunk 2 above)
Comment (hitech95) [reply variants/STM32G0xx/G030C(6-8)T/generic_clock.c]: I did missed that, sorry :)

File: README.md [hunk 3]
```diff
@@ -293,6 +293,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
+| :yellow_heart: | STM32G030C6<br>STM32G030C8 | Generic Board | *2.2.0* |
```
Comment (fpistm) [inline README.md:296]:
```suggestion
| :yellow_heart: | STM32G030C6<br>STM32G030C8 | Generic Board | **2.2.0** |
```

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by ABOSTM — APPROVED** [review]
Except @fpistm ' s remarks, LGTM.

**Review by fpistm — APPROVED** [review]

**Comment by fpistm** [comment]
Hi @hitech95 
We do not have a board with this MCU so we cannot test it but as you made some test  I will merge it.
Thanks again for your contribution.

Note: in this case I squash all the commits as they are for the same goal add a new board support.

**Comment by hitech95** [comment]
Ok, got that! Now I know how to contribute for the next time :)

**Pull request merged**